source/common/http/http1/codec_impl.cc

```diff
@@ -474,10 +475,15 @@ void ConnectionImpl::dispatch(Buffer::Instance& data) {
   if (data.length() > 0) {
     for (const Buffer::RawSlice& slice : data.getRawSlices()) {
       total_parsed += dispatchSlice(static_cast<const char*>(slice.mem_), slice.len_);
+      if (HTTP_PARSER_ERRNO(&parser_) != HPE_OK) {
+        break;
+      }
     }
+    dispatchBufferedBody();
```

Comment:
if we hit a parser error above and HTTP_PARSER_ERRNO(&parser_) != HPE_OK don't we break out of the for loop and then fail the assert in dispatchBufferedBody?

Reply:
The ASSERT is avoided due to exceptions.  As we remove exceptions, the ASSERT should help us make sure we end up handling this case correctly.